main.go

```diff
@@ -33,40 +33,49 @@ func computeCLsVsPOI(bkg, sig, obs []float64) (POI, CLs_exp, CLs_obs []float64)
 
 	// Get B-only expectaction and associated toys
 	model_Bonly := modelPrediction(bkg, sig, 0.0)
-	pseudodata_Bonly := make([][]float64, 0)
-	for i := 0; i < Ntoys; i++ {
-		pseudodata_Bonly = append(pseudodata_Bonly, createPseudodata(model_Bonly))
+	pseudodata_Bonly := make([][]float64, Ntoys)
+	for i := range pseudodata_Bonly {
+		pseudodata_Bonly[i] = createPseudodata(model_Bonly)
 	}
 
 	// Start to loop over mu values
 	nPOI := 20
 	POI_start, POI_end := 0.0, 2.0
-	for iPOI := 0; iPOI < nPOI; iPOI++ {
+
+	POI = make([]float64, nPOI)
+	CLs_exp = make([]float64, nPOI)
+	CLs_obs = make([]float64, nPOI)
+
+	var (
+		nllr_sb = make([]float64, Ntoys)
+		nllr_b  = make([]float64, Ntoys)
+	)
+
+	for i := range POI {
 
 		// Get S+B expectations
-		mu := POI_start + (POI_end-POI_start)/float64(nPOI)*float64(iPOI)
-		POI = append(POI, mu)
+		mu := POI_start + (POI_end-POI_start)/float64(nPOI)*float64(i)
+		POI[i] = mu
 		model_SB := modelPrediction(bkg, sig, mu)
 
 		// Get observed nllr for this assumed POI value
 		nllr_obs := NLLR(obs, model_SB, model_Bonly)
 
 		// Draw some toys to get PDF(nllr|S+B) and PDF(nllr|B)
-		var nllr_sb, nllr_b []float64
-		for i := 0; i < Ntoys; i++ {
-			nllr_sb = append(nllr_sb, NLLR(createPseudodata(model_SB), model_SB, model_Bonly))
-			nllr_b = append(nllr_b, NLLR(pseudodata_Bonly[i], model_SB, model_Bonly))
+		for j := range nllr_sb {
+			nllr_sb[j] = NLLR(createPseudodata(model_SB), model_SB, model_Bonly)
+			nllr_b[j] = NLLR(pseudodata_Bonly[j], model_SB, model_Bonly)
 		}
-		CLs_exp = append(CLs_exp, computeCLs(nllr_sb, nllr_b, stat.Mean(nllr_b, nil)))
-		CLs_obs = append(CLs_obs, computeCLs(nllr_sb, nllr_b, nllr_obs))
+		CLs_exp[i] = computeCLs(nllr_sb, nllr_b, stat.Mean(nllr_b, nil))
+		CLs_obs[i] = computeCLs(nllr_sb, nllr_b, nllr_obs)
 	}
 
 	return POI, CLs_exp, CLs_obs
 }
 
 func modelPrediction(bkg, sig []float64, mu float64) []float64 {
 	prediction := make([]float64, len(bkg))
-	for i := 0; i < len(bkg); i++ {
+	for i := range prediction {
```

Comment:
Why this is not

```go
for i, _ := range prediction 
```
?

Compared to 

```go
for _, val := range x
```

Reply:
when you're only interested in the index of the slice, using the underscore trick isn't needed.
but when you're interested in only the value, you need to tell the compiler that what you want is the second value of the tuple `(index, value)`, hence the `_` for the first value.